### PR TITLE
nixVersions.nix_2_19: init at 2.19.1, nixVersions.unstable: 2.18 -> 2.19

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/package-management/nix/default.nix [hunk 1]
```diff
@@ -235,7 +239,7 @@ in lib.makeExtensible (self: ({
 
   stable = addFallbackPathsCheck self.nix_2_18;
```
Comment (Kranzes) [inline pkgs/tools/package-management/nix/default.nix:240]:
```suggestion
  stable = addFallbackPathsCheck self.nix_2_19;
```

Comment (SuperSandro2000) [reply pkgs/tools/package-management/nix/default.nix:240]: We can bump stable in a bit after determining that things are working as they should.

**Comment by SuperSandro2000** [comment]
@ofborg build nixVersions.nix_2_19

**Comment by happysalada** [comment]
There is a fatal error on darwin that doesn't look great
```
       >   GEN    src/libexpr/primops/derivation.nix.gen.hh
       >   GEN    src/libexpr/fetchurl.nix.gen.hh
       >   CXX    src/libexpr/eval.o
       > src/libexpr/eval.cc:519:18: fatal error: 'flake/call-flake.nix.gen.hh' file not found
       >         #include "flake/call-flake.nix.gen.hh"
       >                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > 1 error generated.
       > make: *** [mk/patterns.mk:3: src/libexpr/eval.o] Error 1
       > rm src/nix/doc/files/profiles.md
       For full logs, run 'nix log /nix/store/cwhq42qdbwp2wxyqanf7q79dqsvmzpp0-nix-2.19.1.drv'.
```

**Comment by trofi** [comment]
I think the build failure is no `darwin`-specific and is a general incomplete dependency problem. Proposed an upstream fix as https://github.com/NixOS/nix/pull/9462

**Comment by trofi** [comment]
`2.19.2` was just released with a fix: https://github.com/NixOS/nix/compare/2.19.1...2.19.2

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 269439` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>nil</li>
    <li>nixVersions.nix_2_19</li>
    <li>nixVersions.nix_2_19.dev</li>
    <li>nixVersions.nix_2_19.doc</li>
    <li>nixVersions.nix_2_19.man</li>
  </ul>
</details>

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 269439` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>nil</li>
    <li>nixVersions.nix_2_19</li>
    <li>nixVersions.nix_2_19.debug</li>
    <li>nixVersions.nix_2_19.dev</li>
    <li>nixVersions.nix_2_19.doc</li>
    <li>nixVersions.nix_2_19.man</li>
  </ul>
</details>

**Review by trofi — APPROVED** [review]

**Pull request merged**